packages/pricing/src/models/price-rule.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  BeforeCreate,
+  Collection,
+  Entity,
+  ManyToMany,
+  ManyToOne,
+  OneToMany,
+  OneToOne,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+
+import MoneyAmount from "./money-amount"
+import PriceSet from "./price-set"
+import PriceSetMoneyAmount from "./price-set-money-amount"
+import RuleType from "./rule-type"
+import { generateEntityId } from "@medusajs/utils"
+
+type OptionalFields = "id" | "is_dynamic" | "priority" 
+type OptionalRelations = "price_set" | "rule_type" | "price_set_money_amount"
+@Entity()
+export default class PriceRule {
+  [OptionalProps]: OptionalFields | OptionalRelations
+  @PrimaryKey({ columnType: "text" })
+  id!: string
+
+
+  @ManyToOne({ entity: () => PriceSet, fieldName: "price_set_id", name: "price_rule_price_set_id_unique"  })
+  price_set: PriceSet
+
+  @ManyToOne({ entity: () => RuleType, fieldName: "rule_type_id", name: "price_rule_rule_type_id_unique" })
+  rule_type: RuleType
+
+  @Property({ columnType: "boolean", default: false })
+  is_dynamic: boolean
+
+  @Property({ columnType: "text" })
+  value: string
+  
+  @Property({ columnType: "integer", default: 0 })
+  priority: number
+
+  @ManyToOne({ entity: () => PriceSetMoneyAmount, fieldName: "price_set_money_amount_id", name: "price_set_money_amount_id_unique" })
+  price_set_money_amount: PriceSetMoneyAmount
+  
+  @Property({ columnType: "text" })
+  price_list_id!: string
+
+  // TODO: Add price list
```

Comment:
question: will it be part of a different pr?

Reply:
supposedly, atm not part of this pr, price list functionality is also not the focus of developing the pricing module at the moment. Keeping this until we eventually start messing with the interactions of price-sets and price lists would be the better approach to not overcomplicate something that's already pretty complicated 😅